internal/executor/executor.go

```diff
@@ -40,3 +42,57 @@ func (e *defaultExecutor) RunCommand(executable string, args []string, cwd util.
 	}
 	return stdout.Bytes(), stderrBuf.Bytes(), err
 }
+
+// createTempRScript creates a temporary file with R code and returns the file path.
+// The caller is responsible for removing the temporary file.
+func createTempRScript(content string) (string, func(), error) {
+	tempFile, err := os.CreateTemp("", "*")
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to create temporary R script: %w", err)
+	}
```

Comment:
I would use defer here to ensure that the file is always deleted instead of handling it within the error blocks. https://go.dev/blog/defer-panic-and-recover